app/controllers/api/v1/owners_controller.rb

```diff
@@ -14,18 +14,23 @@ def show
   def create
     owner = User.find_by_name(params[:email])
     if owner
-      @rubygem.ownerships.create(user: owner)
-      render plain: "Owner added successfully."
+      ownership = @rubygem.ownerships.new(user: owner, authorizer: @api_user)
+      if ownership.save
+        Mailer.delay.ownership_confirmation(ownership.id)
+        render plain: "Owner added successfully. A confirmation mail has been sent to #{owner.handle}'s email"
+      else
+        render plain: ownership.errors.full_messages.to_sentence, status: :unprocessable_entity
+      end
     else
       render plain: "Owner could not be found.", status: :not_found
     end
   end
 
   def destroy
-    owner = @rubygem.owners.find_by_name(params[:email])
+    owner = @rubygem.owners_including_unconfirmed.find_by_name(params[:email])
     if owner
-      ownership = @rubygem.ownerships.find_by(user_id: owner.id)
-      if ownership&.safe_destroy
+      ownership = @rubygem.ownerships_including_unconfirmed.find_by(user_id: owner.id)
+      if ownership.destroy_and_notify
```

Comment:
Sorry if I forgot this.. but was it my suggestion to remove `&` from here? what happens when ownership is nil?

Reply:
No, I did this because, 
`owner = @rubygem.owners_including_unconfirmed.find_by_name(params[:email])` this search for owner from our list of unconfirmed ownerships. So ownership being nil case was not possible here I think. Correct?

Reply:
> onwership being nil would be our db inconsistency

Bold of you to assume it isn't :laughing: Assume that every detail is there for a reason by default. Welcome to legacy.
```
> Ownership.where(user_id: nil).count
=> 7042
```

Reply:
Lol okay :speak_no_evil: